src/utils.rs

```diff
@@ -8,57 +10,89 @@ use serenity::model::{
 use std::collections::HashMap;
 
 pub fn get_num_guilds(guilds: &Vec<GuildStatus>) -> String {
-    if guilds.len() == 1 {
-        String::from("1 guild")
-    } else {
-        format!("{:?} guilds", guilds.len())
-    }
+    format!(
+        "{} guild{}",
+        guilds.len(),
+        if guilds.len() > 1 { "" } else { "s" }
+    )
 }
 
 pub fn get_guild_human_members(members: &Vec<Member>) -> usize {
-    let human_members: Vec<Member> = members
-        .iter()
-        .filter(|member| !member.user.bot)
-        .map(|x| x.clone())
-        .collect();
-
-    human_members.len()
+    members.iter().filter(|member| !member.user.bot).count()
 }
 
 pub fn get_guild_online_members(
     members: &HashMap<UserId, Member>,
     presences: &HashMap<UserId, Presence>,
 ) -> usize {
-    let mut online_members: usize = 0;
-    for (user_id, presence) in presences.iter() {
-        if !members.get(user_id).expect("User not found").user.bot
-            && presence.status == OnlineStatus::Online
-        {
-            online_members += 1;
-        }
-    }
-
-    online_members
+    presences
+        .iter()
+        // Get all online user ids
+        .filter_map(|(user_id, presence)| match presence.status {
+            OnlineStatus::Online => Some(user_id),
+            _ => None,
+        })
+        // Remove members that are bots
+        .filter(|user_id| !members.get(user_id).expect("User not found").user.bot)
+        .count()
 }
 
 pub fn get_text_channels(channels: &HashMap<ChannelId, GuildChannel>) -> usize {
-    let mut text_channels: Vec<GuildChannel> = Vec::new();
-    for channel in channels.values() {
-        if channel.kind == ChannelType::Text {
-            text_channels.push(channel.clone());
-        }
-    }
-
-    text_channels.len()
+    get_channels(channels, ChannelType::Text)
 }
 
 pub fn get_voice_channels(channels: &HashMap<ChannelId, GuildChannel>) -> usize {
-    let mut voice_channels: Vec<GuildChannel> = Vec::new();
-    for channel in channels.values() {
-        if channel.kind == ChannelType::Voice {
-            voice_channels.push(channel.clone());
-        }
+    get_channels(channels, ChannelType::Voice)
+}
+
+pub fn get_channels(channels: &HashMap<ChannelId, GuildChannel>, kind: ChannelType) -> usize {
+    channels
+        .values()
+        .filter(|channel| channel.kind == kind)
+        .count()
+}
+
+static SUBREDDIT_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^/?(r/.+)").expect("Failed to initialise regex pattern"));
```

Comment:
`Lazy` allows you to define a function that's executed once when the value is first used, and then cache the output for later reuse. This is better than compiling your regex in a loop in `main.rs`.